### PR TITLE
Rpg-overhaul

### PR DESCRIPTION
**Pull request by HyperSoop**
Adds RelicEx and Weapon Leveling, also adjusts the playerex exp cost formula to make overlevelling more difficut (you still can level indefinitely with relicex tomes).

RelicEx adds relics that drop from mobs and chests that work as equipment and trinkets and affect the PlayerEx stats. Also adds orbs to refund skill points and tomes to gain those.

I'm yet to do packwiz refresh on this

### CODE REVIEW

**Comment by HyperSoop** [comment]
Packwiz refresh done! I can safely go to sleep now without facing the danger of losing track of my thoughts

**Comment by Gaming32** [comment]
You were one of the ones who said we have enough mods.

**Comment by HyperSoop** [comment]
Unlike some other mods in the pack those stay relevant to the gameplay loop (IMO). You always can just not merge this pr if it doesn't meet your vision for the server

**Comment by intergrav** [comment]
> to make overlevelling more difficut

why would you do this literally right after you got to lvl 1000 lmfao

**Comment by HyperSoop** [comment]
> > to make overlevelling more difficut
> 
> why would you do this literally right after you got to lvl 1000 or whatever lmfao

lvl monopoly + it needs to be done from a balancing perspective. It's a compensated by the level tomes, you still can technically get infinite levels with those.

**Comment by HyperSoop** [comment]
for the sake of my sanity, will this be merged or not, and if not, what do i need to change

**Comment by HyperSoop** [comment]
for the reviewers: do note that levelling can still continue indefinitely and as quickly, if not faster, thanks to relicex's tomes (the give 1 flat levels per and drop from mobs with a like, 0.03% chance)

File: config/yosbr/config/playerex.json
Comment (kuylar) [inline config/yosbr/config/playerex.json:6]:
```suggestion
  "levelFormula": "stairs(x,0.2,2.4,17,10,25)",
```

This should stay as default, since it's been a month the server started, and this change *will* affect both the existing and the newcoming players. You don't see an MMORPG going "hey were suddenly changing our hidden formulas so the game is harder for new players now! oh btw there's a small chance youll get an item drop to instantly level up!"

**Review by kuylar — CHANGES_REQUESTED** [review]

**Comment by Gaming32** [comment]
The one issue is that you are over-leveled like nobody else. This would make it harder for us to reach you. It already takes a school day of AFK XP farming to gain 100 levels under nearly ideal circumstances, and I'm at least 800 levels less than you (and I may very well have the second-most levels).

File: config/yosbr/config/playerex.json
Comment (HyperSoop) [reply config/yosbr/config/playerex.json:6]: If you'd looked at how the graph looks you wouldn't be saying that. The addition does not have a big effect at the lower levels.

**Comment by HyperSoop** [comment]
> The one issue is that you are over-leveled like nobody else. This would make it harder for us to reach you. It already takes a school day of AFK XP farming to gain 100 levels under nearly ideal circumstances, and I'm at least 800 levels less than you (and I may very well have the second-most levels).

This is not a valid argument against the changes. I am fine with my levels being put back to 250 or something with those changes.

**Comment by HyperSoop** [comment]
Again, the addtion to the graph is VERY lenient.  It only reaches 40 levels per at level ~550.

**Comment by Gaming32** [comment]
> > The one issue is that you are over-leveled like nobody else. This would make it harder for us to reach you. It already takes a school day of AFK XP farming to gain 100 levels under nearly ideal circumstances, and I'm at least 800 levels less than you (and I may very well have the second-most levels).
> 
> This is not a valid argument against the changes. I am fine with my levels being put back to 250 or something with those changes.

If this is possible, then I'm ok with that. I'll do an actual review later, when I have more time.

File: config/yosbr/config/playerex.json
Comment (Gaming32) [reply config/yosbr/config/playerex.json:6]: Where can I see a graph?

**Review by Gaming32 — CHANGES_REQUESTED** [review]
Weapon Leveling is on CurseForge, so it should be added as CurseForge mod (`packwiz cf add`). However, it appears to be All Rights Reserved, so you need explicit permission from the mod author for it to be included here. We don't want a DMCA takedown.

**Comment by HyperSoop** [comment]
> Weapon Leveling is on CurseForge, so it should be added as CurseForge mod (`packwiz cf add`). However, it appears to be All Rights Reserved, so you need explicit permission from the mod author for it to be included here. We don't want a DMCA takedown.

https://modrinth.com/mod/weapon-leveling

**Comment by Gaming32** [comment]
> > Weapon Leveling is on CurseForge, so it should be added as CurseForge mod (`packwiz cf add`). However, it appears to be All Rights Reserved, so you need explicit permission from the mod author for it to be included here. We don't want a DMCA takedown.
> 
> https://modrinth.com/mod/weapon-leveling

Please add it as a Modrinth mod then.

**Comment by HyperSoop** [comment]
Wait, there has actually been an update to it two days ago. I might be able to get it working without the stupid ass mixin hack now.

File: config/yosbr/config/playerex.json
Comment (HyperSoop) [reply config/yosbr/config/playerex.json:6]: image
In desmos maybe, this screenshot does not account for the initial 0-25 roll-in (desmos can't do the stairs function smh) so the Y values are gonna be a bit higher than in reality tho

**Review by jadelily18 — APPROVED** [review]
1.19.2 go brrrr

**Comment by HyperSoop** [comment]
I'll work on 1.19.2 whenever upstream stabilizes

**Comment by Gaming32** [comment]
We're gonna wait a few weeks before attempting 1.19.2 again, and we're gonna start from scratch, so don't worry about that.

**Comment by HyperSoop** [comment]
@Gaming32 then i guess this is seldom fine to merge

**Comment by HyperSoop** [comment]
...please someone review this, this is so dead... we could have been done with this a week ago

**Review by intergrav — APPROVED** [review]
t

**Comment by intergrav** [comment]
i thought i did review but i guess i didnt lmao

File: config/yosbr/config/playerex.json
Comment (jadelily18) [reply config/yosbr/config/playerex.json:6]: wouldn't this still make it more difficult for everyone else to get to the level you're at?

Comment (HyperSoop) [reply config/yosbr/config/playerex.json:6]: the tomes

Comment (jadelily18) [reply config/yosbr/config/playerex.json:6]: > the tomes

that really doesn't answer my question

Comment (HyperSoop) [reply config/yosbr/config/playerex.json:6]: The tomes' drop rate is affected by luckiness. Say you have an xp farm that gives you 80 endermen per minute and takes around 5 minutes to get 25 levels at; with luckiness 50 which you'd probably have by the point the increasing level costs start to show you'd have an around 0.1% chance to get a tome from an endermen, that is, around 1000 endermen per tome, which'd be something you'd get every 12-14 minutes. Add this to the levels you get from XP and it compensates out. In addition, you'd still have potential to increase your luckiness and to expand your farm to output more mobs, and that way levelling would be even faster than now.

Comment (HyperSoop) [reply config/yosbr/config/playerex.json:6]: i hope that gets the point across, it's just another meta with more variety and more potential for expansion

**Comment by HyperSoop** [comment]
at this point I'm pretty sure gaming is trying to metagame me by putting this pr on hold indefinitely

**Comment by Gaming32** [comment]
Stop asking for a review. I will give one when I am ready.

**Comment by HyperSoop** [comment]
Gaming you are very clearly intentionally delaying this PR for your own unclear gameplay benefit, otherwise you would have given it a review 10 days ago or so, it can't just be "thinking" at this point. In addition, you always refuse to elaborate on the problems you have with it. This is abuse of power. That's what I think.

If you want to prove me wrong, please collect your thoughts and let's get this merged today.

**Comment by Gaming32** [comment]
As I have said, I will give one when I am ready. Believe it or not, I do in fact have a life outside of Modrinth SMP and your Pull Request.

**Comment by Gaming32** [comment]
What does Weapon Leveling do again?

**Comment by Gaming32** [comment]
What does Weapon Leveling do again? @HyperSoop

**Comment by HyperSoop** [comment]
it makes weapons (and probably armor) level up, increasing their stats

**Review by Gaming32 — APPROVED** [review]

**Review by jadelily18 — APPROVED** [review]

**Comment by HyperSoop** [comment]
The levelup cost formula could be just changed to "x+1" and it'd be well balanced and not too grindable; with relicex's orbs of regret you can even redistribute your points after putting them in.

"0.5*x+0.1" could also be done which would basically be the golden middle where you can get quite a bit of benefits but can't overgrind.

**Pull request merged**
